src/Operators/finitedifference.jl

```diff
@@ -1320,6 +1320,204 @@ Base.@propagate_inbounds function stencil_right_boundary(
     stencil_interior(op, loc, space, idx - 1, hidx, velocity, arg)
 end
 
+struct LinVanLeerC2F{BCS} <: AdvectionOperator
+    bcs::BCS
+end
+
+"""
+    LimiterConstraint
+Following the van Leer class of limiters as noted in 
+[Lin1994](@cite), four limiter constraint options are provided
+- AlgebraicMean: Algebraic mean, this guarantees neither positivity nor monotonicity (eq 2)
+- PosDef: Positive-definite with implicit diffusion based on local stencil extrema (eq 3b, 3c, 5a, 5b)
+- Mono4: Monotonicity preserving harmonic mean, this implies a strong monotonicity constraint (eq 4)
+- Mono5: Monotonicity preserving, with extrema bounded by the edge cells in the stencil (eq 5)
+
+The diffusion implied by these methods is proportional to the local upwind CFL number.
+The `mismatch` Δ𝜙 = 0 returns the first-order upwind method. Special cases (discussed in Lin et al (1994)) include setting the 𝜙_min = 0 or 𝜙_max = saturation mixing ratio for water vapor
+are not considered here in favour of the generalized local extrema in equation (5a, 5b).
+"""
+abstract type LimiterConstraint end
+struct AlgebraicMean <: LimiterConstraint end
+struct PosDef <: LimiterConstraint end
+struct Mono4 <: LimiterConstraint end
+struct Mono5 <: LimiterConstraint end
+
+LinVanLeerC2F(; kwargs...) = LinVanLeerC2F(NamedTuple(kwargs))
+
+return_eltype(::LinVanLeerC2F, V, A, dt) =
+    Geometry.Contravariant3Vector{eltype(eltype(V))}
+
+return_space(
+    ::LinVanLeerC2F,
+    velocity_space::AllFaceFiniteDifferenceSpace,
+    arg_space::AllCenterFiniteDifferenceSpace,
+    dt,
+) = velocity_space
+
+function compute_Δ𝛼_linvanleer(a⁻, a⁰, a⁺, v, dt, ::Mono5)
+    Δ𝜙_avg = ((a⁰ - a⁻) + (a⁺ - a⁰)) / 2
+    min𝜙 = min(a⁻, a⁰, a⁺)
+    max𝜙 = max(a⁻, a⁰, a⁺)
+    𝛼 = min(abs(Δ𝜙_avg), 2 * (a⁰ - min𝜙), 2 * (max𝜙 - a⁰))
+    Δ𝛼 = sign(Δ𝜙_avg) * 𝛼 * (1 - sign(v) * v * dt)
+end
+
+function compute_Δ𝛼_linvanleer(a⁻, a⁰, a⁺, v, dt, ::Mono4)
+    Δ𝜙_avg = ((a⁰ - a⁻) + (a⁺ - a⁰)) / 2
+    c = sign(v) * v * dt
+    if sign(a⁰ - a⁻) == sign(a⁺ - a⁰) && Δ𝜙_avg != eltype(v)(0)
```

Comment:
The conversion to `eltype(v)` on this line is unnecessary (it will be inserted automatically by the compiler) and makes the code a bit harder to read. The only places where we need such conversions are isolated return statements (e.g., `return eltype(v)(0)` a few lines down). You can remove all such conversions from conditional statements like this one.

Reply:
Noted, fixed the important ones.